internal/controllers/reconciliation/controller.go

```diff
@@ -136,12 +147,17 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 
 func (c *Controller) reconcileResource(ctx context.Context, prev, resource *reconstitution.Resource, current *unstructured.Unstructured) (bool, error) {
 	logger := logr.FromContextOrDiscard(ctx)
+	start := time.Now()
+	defer func() {
+		reconciliationLatency.Observe(time.Since(start).Seconds())
```

Comment:
Nice, I completely forgot that arguments of deferred functions are evaluated immediately